packages/affiliates/libs/uniswap/utils.ts

```diff
@@ -32,3 +24,58 @@ export function convertValuesToString<T>(obj: { [k: string]: any }): T {
     })
   ) as T;
 }
+
+// This looks at a positions liquidity, and calculates the amount across a single tick based on full tick span
+export function liquidityPerTick(params: { liquidity: string; tickLower: string; tickUpper: string }) {
+  const { liquidity, tickLower, tickUpper } = params;
+  assert(BigInt(tickUpper) > BigInt(tickLower), "Upper tick must be > lower tick");
+  return (BigInt(liquidity) / (BigInt(tickUpper) - BigInt(tickLower))).toString();
+}
+
+// this is stronger than position, but compatible, since these fields have been validated to exist
+type ActivePosition = Position & {
+  liquidity: BigNumberish;
+  tickUpper: BigNumberish;
+  tickLower: BigNumberish;
+};
+
+// figure out if a position is active by looking a liquidity and combination of current tick and position bounds
+// this assume the upper tick is not inclusive, meaning if the upper tick matches the current tick, its not active
```

Comment:
is this assumption true? if the upper tick is the current tick I thought it _was_ active?

Reply:
o, wait. it might just be inclusive with the lower tick 🤔 . in any case can you provide a bit more on this assumption?